common_knowledge/GitBranching.md

```diff
@@ -0,0 +1,45 @@
+
+# Git Branching
+
+We use a model something close to [Trunk-based development](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) 
+where we try to merge code to master often and send in small pull requests.
+
+We use the standard/default merge strategy for Git which avoids helps avoid
+merge conflicts so you can merge master and smaller (possibly stacked) PRs into
+your development branch often.
+
+# Key Suggestions
+
+## Commit early, commit often
+
+Make sure to commit your code early and often. This allows for smaller and
+discrete integration tests but also reduces the chance that your code will cause
+a merge conflict later.
+
+## Conventional Commits
+
+Use [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) in your PR title.
```

Comment:
how are we gonna utilise conventional commits? For changelog? Also, if we decide to go this way, probably it should be used on every commit, not only on PR title

Reply:
Yeah I also agree here, we should be first asking ourselves, what is the value of the added complexity.

If it is just for semantic versioning, there will be an automated way of openapi spec diffs:
https://github.com/OpenAPITools/openapi-diff

We can use this tool to automate versioning, this would have the added benefit of not being error prone (for example someone marks their commit as something else, but it is really a breaking change)

Reply:
Fortunately, @kurtisassad @masvelio We only require them on the PR title. You *may* add them to each individual commit if you want but the real git history is PR to PR and git log --merged will show this.

It should be somewhat easy to implement.

Reply:
If you "may" add them, there is no mention of it being optional in the doc.

If we want conventional commits in the PR title, how can we enforce it? Usually you can use a precommit tool if you are doing it every commit, but since we only require it being in the title this strategy won't work.

It may be "easy" to implement but without it being enforced by some automated process there will be many slipping through without the conventional commit style.

But I still think we are getting ahead of ourselves here. We should evaluate what benefits conventional commits give us, as there is still no discussion of it.

Reply:
@kurtisassad Sorry for not clarifying. There's a few github actions that do this on the PR title.  I removed the conventional commits from this documentation though and we can discuss it separately.